### DISCUSSION TITLE
Support postgresql pgpassfile for authentication

### DISCUSSION THREAD

**Discussion by gjacquet** [grafana/grafana#61516] [Platform]
I have some use cases where it would be more convenient to use postgresql pgpass (https://www.postgresql.org/docs/current/libpq-pgpass.html) for authentication. One of them includes using dynamic credentials with IAM RDS authentication but more generally this would allow updating credentials without restarting grafana.

Please that this request applies to the Grafana's SQL store but it would be just as interesting to have the same support for postgres datasource.

[lib/pq](https://github.com/lib/pq) already has support for pgpass built-in. It would require very little change in Grafana. It would only require to generate a connection string that does not include the password parameter at all when password is empty for lib/pq to use credentials from pgpass file if it exists.

I did some tests and implementations looks pretty straightforward.

**Comment by gjacquetupgrade** [comment]
I created https://github.com/grafana/grafana/pull/61517 as a sample implementation.
